### PR TITLE
Add Amber framework.

### PR DESCRIPTION
**Pull request by faustinoaq**
This PR adds Amber framework.

 [Amber](https://www.ambercr.io) is a web application framework written in [Crystal](http://www.crystal-lang.org) inspired by Kemal, Rails, Phoenix and other popular application frameworks.

amber

### CODE REVIEW

File: frameworks/Crystal/amber/shard.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+license: MIT
+
+dependencies:
+  amber:
+    github: Amber-Crystal/amber 
+    version: ~> 0.1.4
+
+  granite_orm:
+    github: Amber-Crystal/granite-orm
+    version: ~> 0.6.2
```
Comment (drujensen) [inline frameworks/Crystal/amber/shard.yml]: @faustinoaq We will want the latest version before this gets merged.

Comment (faustinoaq) [inline frameworks/Crystal/amber/shard.yml]: @drujensen I just see this PR https://github.com/Amber-Crystal/granite-orm/pull/39 :+1: 

Do I need to change version to master? or a new version of granite-orm will be released?

Comment (drujensen) [inline frameworks/Crystal/amber/shard.yml]: @faustinoaq I pushed a new release `0.6.3`.  I recommend removing the version from the shard.yml, perform a `shard update` and then check in the `shard.lock` file so it doesn't get updated.

Comment (faustinoaq) [inline frameworks/Crystal/amber/shard.yml]: @drujensen done :+1:

**Comment by faustinoaq** [comment]
@nbrady-techempower I think this PR is working well, Could Amber be on round 15?

File: frameworks/Crystal/amber/setup.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+fw_depends postgresql crystal
+
+shards install --production
+
+crystal build --release --no-debug src/amber.cr
+
+for i in $(seq 1 $((2 * $(nproc --all)))); do
```
Comment (RX14) [inline frameworks/Crystal/amber/setup.sh]: I'm not sure launching 2 processes per core will gain anything on the actual benchmark machine. I think it'd be best to actually benchmark this on the target machine. Excess process switches are bad.

**Comment by drujensen** [comment]
@RX14 I found a small performance improvement since there seems to be some io wait time in the other processes, however I agree that `2 *` is excessive.  Adding just a couple more processes above the number of cpu's will suffice.

**Comment by RX14** [comment]
@drujensen How many cores are you testing on locally?

**Comment by drujensen** [comment]
@RX14 I tested on 2 cores but admittedly in a VM.  Maybe not a valid test?

**Comment by faustinoaq** [comment]
Yes, then 1 process per core would be better?, I tested in my computer and
always the best result was when I executed the same amount as process as
cores.

**Comment by drujensen** [comment]
ok, go ahead and remove the `2 *`.  Thanks.

**Comment by RX14** [comment]
I don't think testing on 2 cores is very valid when the actual benchmark setup has 80 cores. At 80 processes, at least 1 of them will likely be running a GC at 1 time, and the GC runs concurrently. This means that i hypothesize that the ideal number of threads will be less than the number of cores. Unfortunately I don't have the benchmark machine to test that on.

**Comment by drujensen** [comment]
@RX14 right, agree.

**Comment by faustinoaq** [comment]
@drujensen Do you will update Kemalyst too?

**Comment by nbrady-techempower** [comment]
Hey guys, let me know when you feel like this is ready to merge.

**Comment by faustinoaq** [comment]
I tested it in my local machine without problems.

Also, all dependencies are updated.

@nbrady-techempower Could Amber be included on round 15?

@drujensen and @RX14, Do you think this PR is ready?

**Comment by drujensen** [comment]
@nbrady-techempower @faustinoaq LGTM.  This is ready to be merged.  Thanks!

**Comment by nbrady-techempower** [comment]
Thanks guys. @drujensen I'll get this in now and it should be included in Preview 3. It's been extremely busy in the office but we should have Preview 2 out any day.

**Comment by drujensen** [comment]
@nbrady-techempower :+1:  Thanks!

**Pull request merged**